extensions/schema-compare/src/schemaCompareResult.ts

```diff
@@ -311,15 +316,27 @@ export class SchemaCompareResult {
 			let key = diff.sourceValue ? diff.sourceValue : diff.targetValue;
 			if (key) {
 				if (!this.sourceTargetSwitched) {
-					delete this.originalSourceExcludes[key];
+					this.originalSourceExcludes.delete(key);
 					if (!rowState.checked) {
-						this.originalSourceExcludes[key] = diff;
+						this.originalSourceExcludes.set(key, diff);
+						if (this.originalSourceExcludes.size === this.comparisonResult.differences.length) {
+							this.toggleNoChanges(false);
+						}
+					}
+					else if (!this.generateScriptButton.enabled && !this.applyButton.enabled) {
```

Comment:
nit: these checks should be unnecessary - wouldn't you always want to enable the buttons? It may set enabled for buttons that are already enabled but that's fine.